airbyte-cdk/java/airbyte-cdk/typing-deduping/src/main/java/io/airbyte/integrations/base/destination/typing_deduping/TyperDeduperUtil.kt

```diff
@@ -0,0 +1,72 @@
+package io.airbyte.integrations.base.destination.typing_deduping
+
+import com.google.common.collect.Streams
+import io.airbyte.cdk.integrations.util.ConnectorExceptionUtil.getResultsOrLogAndThrowFirst
+import io.airbyte.commons.concurrency.CompletableFutures
+import java.util.*
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.CompletionStage
+import java.util.concurrent.ExecutorService
+
+
+class TyperDeduperUtil {
+  companion object {
+
+    @JvmStatic
+    fun executeRawTableMigrations(
+        executorService: ExecutorService,
+        sqlGenerator: SqlGenerator,
+        destinationHandler: DestinationHandler,
+        v1V2Migrator: DestinationV1V2Migrator,
+        v2TableMigrator: V2TableMigrator,
+        parsedCatalog: ParsedCatalog
+    ) {
+      // TODO: Either the migrations run the soft reset and create v2 tables or the actual prepare tables.
+      // unify the logic
+      // with current state of raw tables & final tables. This is done first before gather initial state
+      // to avoid recreating
+      // final tables later again.
+      val runMigrationsResult =
```

Comment:
as noted above - this code originally lived in DefaultTyperDeduper, I just extracted it to a separate class.